src/pages/Search.js

```diff
@@ -1,16 +1,45 @@
-import React from 'react';
-import { useParams } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { Redirect, useParams, withRouter } from 'react-router-dom';
+import SubredditForm from '../components/SubredditForm';
 import defaultSubreddit from '../helpers/constants';
 
+
 const Search = () => {
-  const { subreddit = defaultSubreddit } = useParams();
-  return (
+  const { subredditQuery } = useParams();
+  const [subreddit, setSubreddit] = useState(subredditQuery || defaultSubreddit);
+  const [route, setRoute] = useState(null);
+
+  const handleSubmit = (evt) => {
+    setRoute(`/search/${subreddit}`);
+    evt.preventDefault();
+  };
+
+  useEffect(() => {
+    document.getElementById('subreddit').value = subredditQuery;
```

Comment:
Ah now I understand why you set the id in the form :smile: Accessing the DOM directly via `document.getElementById` or `document.querySelector` is in most cases a code-smell though. Often there's a simpler solution that involves state, sometimes you need `useRef`.

In this case, you can set the `subreddit` state directly with `setSubreddit` :wink:

Reply:
dear god... it was as simple as using the useHistory hook... that was the piece I was missing.  I tried so many different ways to get this to work and ended up with the document.get...  It didn't feel right but I wasn't exactly sure why!  thank you!